### DISCUSSION TITLE
How to use ResNet API?

### DISCUSSION THREAD

**Discussion by markvvw** [Project-MONAI/MONAI#5715] [Q&A] [answered]
Recently, I attempted to use ResNet for building a classification model of 3D MRI (256*256*16). However, I do not know how to set the parameters as a medical doctor. There were no correct information for addressing this problem.

 Thank you in advance for your reply.

FALSE: 
model = ResNet(layers=[18], spatial_dims=3,num_classes=2, n_input_channels=1,block='basic',block_inplanes=[2, 2, 2, 2]).to(device)#

**Comment by KumoLiu** [comment, marked as answer]
Hi @markvvw, you can try `resnet18`, `resnet50` or other resnet in this [file](https://github.com/Project-MONAI/MONAI/blob/dev/monai/networks/nets/resnet.py), just replace `num_classes`.
```
from monai.networks.nets import resnet18
model = resnet18(num_classes=3)
```
Hope it can help you, thanks!

**Reply by markvvw** [reply]
感谢，我就不用英语写了 ：）

**Reply by markvvw** [reply]
> Hi @markvvw, you can try `resnet18`, `resnet50` or other resnet in this [file](https://github.com/Project-MONAI/MONAI/blob/dev/monai/networks/nets/resnet.py), just replace `num_classes`.
> 
> ```
> from monai.networks.nets import resnet18
> model = resnet18(num_classes=3)
> ```
> 
> Hope it can help you, thanks!

resnet18需要修改channel数，它期望的是3， 但我的数据是1，怎么修改啊？

**Reply by KumoLiu** [reply]
Hi @markvvw, you could change the input_channel by setting `n_input_channels`. Thanks!

**Reply by Wangrui-berry** [reply]
Hello, if I want to use 3d resnet, should I change spatial_dims as 3 to change resnet (conv2d) to 3d? 
model = monai.networks.nets.resnet10(spatial_dims=3, n_input_channels=1, n_classes=1)
How can I use conv3d?

**Reply by Wangrui-berry** [reply]
model = resnet18(num_classes=3)
This code is still use conv2d to process 3d data, isn't it?

**Reply by KumoLiu** [reply]
Hi @Sudayberry, yes you need to change `spatial_dims` to 3 if you want to use 3d Resnet but don't need to change conv by yourself. The network will automatically use 3dconv after changing `spatial_dims`.

Hope it can help you, thanks!

**Reply by Wangrui-berry** [reply]
Thanks. That's a good answer to my question.
